src/internal/_Set.js

```diff
@@ -37,11 +37,15 @@ module.exports = (function() {
       case 'string':
       case 'number':
         // distinguish between +0 and -0
-        if (item === 0 && !set._items['-0'] && 1 / item === -Infinity) {
-          if (shouldAdd) {
-            set._items['-0'] = true;
+        if (item === 0 && 1 / item === -Infinity) {
+          if (set._items['-0'] === true) {
+            return !shouldAdd;
+          } else {
+            if (shouldAdd) {
+              set._items['-0'] = true;
+            }
+            return shouldAdd;
           }
```

Comment:
I find the logic above difficult to follow. The `shouldAdd` parameter indicates that this function has two modes. The logic above is confusing because it references this parameter several times. I'd like to see a single branch which reflects the fact that `shouldAdd` is a switch. I suggest something along these lines:

```javascript
var setContainsNegativeZero = set._items['-0'] === true;
if (shouldAdd) {
  return setContainsNegativeZero ? false : (set._items['-0'] = true);
} else {
  return setContainsNegativeZero;
}
```

I would also like to see `{'-0': false}` rather than `{}` as the initial value of `set._items`. This would remove the need for `=== true`, and would indicate—in the constructor—the purpose of the `_items` property.